lib/modules/manager/npm/post-update/npm.ts

```diff
@@ -119,6 +119,10 @@ export async function generateLockFile(
           .map((update) => update.managerData?.packageKey)
           .join(' ');
       commands.push(updateCmd);
+      if (config.packageFile) {
+        commands.push(`git checkout HEAD -- ` + config.packageFile);
```

Comment:
One challenge with this might be that users mix together updates in the same PR - an update before this could have _intentionally_ updated package.json and then this change would undo that change and essentially revert that update in the branch.

Reply:
Should we just switch to npm update fastify without version? Would that have any drawbacks?

Reply:
Everything has a downside unfortunately. In this case it means no control over the version. Eg if someone has minimumReleaseAge configured

Reply:
Is it possible to commit package.json in the npm post-update step or is that "wrong" to do in this step?

Reply:
@rarkins I accidentally stumpled upon some code, that looked exactly like it try todo exactly what I was trying to fix, and it was hardcoded to work only with v2 lockfile.

I added >= 2 and my debug repository now has correct MR's 

https://github.com/firecow/renovate-debug-pull-23495/pulls seen here.